cloudflare/resource_cloudflare_worker_script.go

```diff
@@ -53,26 +43,15 @@ type ScriptData struct {
 }
 
 func getScriptData(d *schema.ResourceData, client *cloudflare.API) (ScriptData, error) {
-	zoneID := d.Get("zone_id").(string)
 	scriptName := d.Get("name").(string)
-	if zoneID == "" && scriptName == "" {
-		return ScriptData{}, fmt.Errorf("either `zone_id` or `name` field must be set")
-	}
 
 	var params cloudflare.WorkerRequestParams
 	var id string
 
-	if scriptName != "" {
-		params = cloudflare.WorkerRequestParams{
-			ScriptName: scriptName,
-		}
-		id = "name:" + scriptName
-	} else {
-		params = cloudflare.WorkerRequestParams{
-			ZoneID: zoneID,
-		}
-		id = "zone:" + zoneID
+	params = cloudflare.WorkerRequestParams{
+		ScriptName: scriptName,
 	}
+	id = "name:" + scriptName
```

Comment:
We could just have the `scriptName` be the ID now that we don't need to differentiate between single-script and multi-script resources. However, I'm not sure what the backwards-compat concerns are around changing the ID format. It will work fine even if we leave the `name:` prefix.

Reply:
Seeing how this is going into 2.0, we can remove this and make a backwards incompatible change as long as we document the change and what people need to do to keep it working. Something we might want to consider though is a way for existing scripts to migrate to this though. In the past I've used [state migrations](https://www.terraform.io/docs/extend/resources/state-migration.html) however I'm not sure if they are different in 0.12.x now.

Reply:
Removed.